components/MDX/Image/index.tsx

```diff
@@ -0,0 +1,22 @@
+import type { ImageProps } from 'next/image';
+import Image from 'next/image';
+import type { DetailedHTMLProps, FC, ImgHTMLAttributes } from 'react';
+
+type HTMLImageProps = DetailedHTMLProps<
+  ImgHTMLAttributes<HTMLImageElement>,
+  HTMLImageElement
+>;
+
+const MDXImage: FC<ImageProps> = ({ width, height, alt, ...props }) => {
+  if (!width || !height) {
+    // When `width` and `height` are not provided in markdown image format, we
+    // have to use the default HTML `img` tag.
+    // @example ![alt](url title)
+    // eslint-disable-next-line @next/next/no-img-element
+    return <img {...(props as HTMLImageProps)} alt={alt} />;
```

Comment:
I assume we won't benefit from Next.js in these situations. Since most Markdown pages will use regular Markdown image tag, and if they use HTML tags they need to be MDX... So, I wonder, is there any way to be able to use images on that situation? Like auto width/height?

Reply:
Let me check 👀